xcodeproj/internal/xcodeproj_rule.bzl

```diff
@@ -189,6 +192,11 @@ targets.
                 additional_linking_files,
             )
 
+        for file, owner_label in owned_extra_files.items():
+            if str(label) == str(owner_label):
+                for f in file.files.to_list():
+                    focused_targets_extra_files.append((None, [file_path(f)]))  # FIXME: should use label instead of None
```

Comment:
I'm not sure what the key value should be here. If I leave it `None`, things seems to work fine, but from the initial issue, it seems like it should instead be the label of the target it should be associated with? If I simply pass `label` the file doesn't get added to the project.

Reply:
It should be the `id` of the target. So we should create a dictionary that maps labels to every id, then here we would append once for each id in that list for the label.

Reply:
I noticed one thing when implementing this (which can be seen in the updated fixtures as well). Due to the fact that one target can map to multiple ids, this can cause some interesting things:
```
//Lib:Lib -> //Lib:Lib ios-sim_arm64-min15.0-applebin_ios-ios_sim_arm64-dbg-ST-8c5deb722afc
//Lib:Lib -> //Lib:Lib watchos-arm64-min7.0-applebin_watchos-watchos_arm64-dbg-ST-9b383575a231
//Lib:Lib -> //Lib:Lib tvos-sim_arm64-min15.0-applebin_tvos-tvos_sim_arm64-dbg-ST-88387ad25201
//Lib:Lib -> //Lib:Lib watchos-arm64-min7.0-applebin_watchos-watchos_arm64-dbg-ST-50c41d527b2
```

Due to the logic to merge targets, is this fine, or do we need to find a different way?

Reply:
This is expected. And it means that the file will be in the project if any of those targets end up in the file. A file is only added once, regardless of how many targets claim "ownership".

Since we currently focus or unfocus based on label, and not id, we could probably just add ownership to the first id in the list. And we could probably go one step further just make the dictionary hold a single id instead of a list of ids. (All of these optimizations with comments, in case that assumption changes in the future).

Reply:
I can add the optimizations in a follow-up.